xtra/src/lib.rs

```diff
@@ -114,16 +113,16 @@ use crate::recv_future::Message;
 ///     })
 /// }
 /// ```
-#[async_trait::async_trait]
 pub trait Handler<M>: Actor {
     /// The return value of this handler.
     type Return: Send + 'static;
 
     /// Handle a given message, returning its result.
-    ///
-    /// This is an [`async_trait`](https://docs.rs/async-trait).
-    /// See the trait documentation to see an example of how this method can be declared.
-    async fn handle(&mut self, message: M, ctx: &mut Context<Self>) -> Self::Return;
+    fn handle(
+        &mut self,
+        message: M,
+        ctx: &mut Context<Self>,
+    ) -> impl Future<Output = Self::Return> + Send;
```

Comment:
A limitation of the current feature is that there is no "return-type notation", meaning users of this trait cannot specify, whether or not this future is `Send`. If we would just use `async fn` here in the declaration, this `Future` would not be `Send` and thus we would not be able to use `xtra` on multi-threaded executors.

From reading https://github.com/rust-lang/rust/pull/115822, this is what is most likely going to be stablized. It is still quite cool that this just works now! :)

Reply:
As we can see in the rest of the codebase, implementations can just use `async fn` and they will inherit the `Send` requirement.